### PR TITLE
REF: shallow_copy->simple_new/rename

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/core/indexes/range.py [hunk 1]
```diff
@@ -169,9 +170,16 @@ def _data(self):
         return np.arange(self.start, self.stop, self.step, dtype=np.int64)
 
     @cache_readonly
-    def _int64index(self) -> Int64Index:
+    def __int64index(self) -> Int64Index:
```
Comment (jreback) [inline pandas/core/indexes/range.py:173]: can you rename these to something else, maybe

__int64index -> _create_int64index
_int64index -> _wrap_int64index

maybe not much better, but

**Review by jreback — CHANGES_REQUESTED** [review]
can you completely remove _shallow_copy?

File: pandas/core/indexes/range.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/indexes/range.py:173]: good call

**Comment by jbrockmendel** [comment]
> can you completely remove _shallow_copy?

Short answer: i think so.

Use cases that are just `self._shallow_copy()` could be replaced with `self.view()` (or implement a `self._view` fastpath) and `self._shallow_copy(name=foo)` that could use `self.rename` (or implement a `self._rename` fastpath).

There's one usage that can't be removed until #38140 (or we implement an equivalent shim in `Categorical._get_unique_index`)

That leaves 7 usages in core.indexes which i think can use simple_new directly (though MultiIndex may throw a wrench in that)

**Review by jreback — APPROVED** [review]

**Comment by jbrockmendel** [comment]
Want to move forward with the rest of the shallow_copy usages?

**Comment by jreback** [comment]
yep i think that would be good

**Pull request merged**